packages/x-components/src/x-modules/query-suggestions/components/__tests__/query-suggestions.spec.ts

```diff
@@ -116,14 +114,33 @@ describe('testing Query Suggestions component', () => {
       store
     });
 
-    const suggestionsItemWrappers = querySuggestions.findAll(
-      getDataTestSelector('query-suggestion')
-    ).wrappers;
+    const suggestionsItemWrappers = findTestDataById(querySuggestions, 'query-suggestion').wrappers;
     expect(suggestionsItemWrappers).toHaveLength(suggestions.length);
 
     suggestionsItemWrappers.forEach((slot, index) => {
       expect(slot.find(getDataTestSelector('icon')).element).toBeDefined();
       expect(slot.find(getDataTestSelector('query')).text()).toEqual(suggestions[index].query);
     });
   });
+
+  // eslint-disable-next-line max-len
+  it('renders at most the number of QuerySuggestion defined by `maxItemsToRender` prop', async () => {
+    resetXQuerySuggestionsStateWith(store, { suggestions });
+
+    const renderedQuerySuggestions = (): WrapperArray<Vue> =>
+      findTestDataById(querySuggestionsWrapper, 'query-suggestion');
+
+    await querySuggestionsWrapper.setProps({ maxItemsToRender: 2 });
+    expect(renderedQuerySuggestions()).toHaveLength(2);
+
+    await querySuggestionsWrapper.setProps({ maxItemsToRender: 3 });
+    expect(renderedQuerySuggestions()).toHaveLength(3);
+
+    await querySuggestionsWrapper.setProps({ maxItemsToRender: 5 });
+    expect(renderedQuerySuggestions()).toHaveLength(suggestions.length);
+  });
```

Comment:
Same as the base-suggestions test, if we don't explicitly set the suggestions in the test, I would parametrise below assertions.

```suggestion
    const renderedQuerySuggestions = (): WrapperArray<Vue> =>
      findTestDataById(querySuggestionsWrapper, 'query-suggestion');

   expect(renderedQuerySuggestions()).toHaveLength(5); // Default value of max items to render

    await querySuggestionsWrapper.setProps({ maxItemsToRender: suggestions.length - 1 });
    expect(renderedQuerySuggestions()).toHaveLength(suggestions.length - 1);

    await querySuggestionsWrapper.setProps({ maxItemsToRender: suggestions.length });
    expect(renderedQuerySuggestions()).toHaveLength(suggestions.length);

    await querySuggestionsWrapper.setProps({ maxItemsToRender: suggestions.length + 1 });
    expect(renderedQuerySuggestions()).toHaveLength(suggestions.length);
  });
```